swatch-metrics/src/main/java/com/redhat/swatch/metrics/util/MeteringEventFactory.java

```diff
@@ -147,6 +149,25 @@ public static void updateMetricEvent(
         .withProductTag(Set.of(productTag))
         .withProductIds(productIds)
         .withConversion(is3rdPartyMigrated);
+
+    BillingProvider billingProviderEnum = getBillingProvider(billingProvider, orgId, instanceId);
+    toUpdate.withBillingProvider(billingProviderEnum);
+
+    if (billingProviderEnum != null) {
+      var cloudProvider = getCloudProvider(billingProviderEnum);
+      cloudProvider.ifPresent(
+          x -> toUpdate.withHardwareType(HardwareType.CLOUD).withCloudProvider(x));
+    }
+  }
+
+  static Optional<CloudProvider> getCloudProvider(BillingProvider billingProvider) {
+
+    return switch (billingProvider) {
+      case AWS -> Optional.of(CloudProvider.AWS);
+      case GCP -> Optional.of(CloudProvider.GOOGLE);
+      case AZURE -> Optional.of(CloudProvider.AZURE);
+      default -> Optional.empty();
+    };
```

Comment:
@lindseyburnett I'm wondering if we should move this into the Event ingestion space since that's what we are doing with the product tags. I'll wait for your feedback in the morning.

Reply:
I'm inclined to keep in swatch-metrics to be consistent with the other Event producers that swatch-tally consumes.  Cost Management sends events with CloudProvider and HardwareType, so I'm thinking we should keep this decoupled and treat swatch-metrics as if it's just another "outside" producer and fill out as much as possible (within reason, limiting to the domain that swatch-metrics own).

Reply:
Sounds good.